ml_pipeline/scripts/training.py

```diff
@@ -0,0 +1,76 @@
+import os
+import json
+import pickle
+import pandas as pd
+import numpy as np
+import rasterio as rio
+import geopandas as gpd
+from pycaret.classification import *
+import argparse
+
+def load_config(config_path):
+    with open(config_path, 'r') as config_file:
+        config = json.load(config_file)
+    return config
+
+def train_with_config(config):
+    gdf = gpd.read_file(config['Paths']['shapefile_path'])
+    new_df1 = gdf.drop(columns=config['ColumnsToDrop'])
+    
+    new_df1 = new_df1.rename(columns=lambda col: extract_number(col))
+
+    # Load global statistics from cache
+    global_stats_dict = {band_name: cache_global_stats(band_name, config['Paths']['pickle_dir']) for band_name in band_names}
```

Comment:
Here, I think that `cache_global_stats` is defined in `prediction.py` but is used here (without being imported). Since it's only used in `training.py`, I think that this would be the best file to define it in.